external/fv3fit/fv3fit/data/tfdataset.py

```diff
@@ -115,6 +115,10 @@ class WindowedZarrLoader(TFDatasetLoader):
         n_windows: number of windows to create per epoch, defaults to the number
             of windows needed to fully cover the dataset with overlapping
             start and end times.
+        time_start_index: index of the first time step of the raw dataset to use,
```

Comment:
I'd like to leave this for now if that's OK, making this some kind of Slice sub-dataclass would leave the larger issues this class has with time selection clarity, as time_stride and n_windows are already a little confusing. It may be better for these time arguments to all be at the same level, at least.

If this gets to be a bigger issue I'll refactor them all out together, or refactor both time slicing and windowing and make it clear how they interact.

Reply:
Sure, I just meant changing the ordering of the optional arguments, rather than making an actual slice argument.

Reply:
Ah, that would actually make it a bit misleading (which is maybe more of a reason for me to separate these options out). The start and end indices happen at the start as pre-subsetting of the dataset, while the time stride is used when selecting windows randomly within that dataset. The distinction is that if you have an hourly dataset, windows will always start on any hour of the day, even if your time stride is 3, as opposed to the dataset only having data every 3 hours before selecting windows.